src/styles/common/_input.scss

```diff
@@ -48,37 +48,54 @@ select {
     position: relative;
     width: calc(map-get($space-sizes, xlarge) + map-get($space-sizes, xlarge));
     height: map-get($space-sizes, xlarge);
-}
 
-.switch input {
-    opacity: 0;
-    width: 0;
-    height: 0;
-}
+    &.disabled {
+        cursor: not-allowed;
+        .slider:before {
+            background-color: map-get($colors, disabled-primary);
```

Comment:
not a huge fan of the grey for the disabled knob. To me still looks like its interactable (but i can be convinced). Can we instead do a opacity disable style like the following?

<img width="460" alt="Screenshot 2024-08-12 at 10 32 56 PM" src="https://github.com/user-attachments/assets/4d8bdb00-3a5e-4c22-87a9-6317d86d98c7">